deepchem/models/torch_models/gat.py

```diff
@@ -0,0 +1,225 @@
+  True
+
+  References
+  ----------
+  .. [1] Veličković, Petar, et al. "Graph attention networks." arXiv preprint
+     arXiv:1710.10903 (2017).
+
+  Notes
+  -----
+  This class requires PyTorch Geometric to be installed.
+  """
+
+  def __init__(
+      self,
+      in_node_dim: int = 38,
+      hidden_node_dim: int = 64,
+      heads: int = 4,
+      dropout_rate: float = 0.0,
+      num_conv: int = 3,
+      predicator_hidden_feats: int = 32,
+      n_tasks: int = 1,
+  ):
+    """
+    Parameters
+    ----------
+    in_node_dim: int, default 38
+      The length of the initial node feature vectors. The 38 is
+      based on `MolGraphConvFeaturizer`.
+    hidden_node_dim: int, default 64
+      The length of the hidden node feature vectors.
+    heads: int, default 4
+      The number of multi-head-attentions.
+    dropout_rate: float, default 0.0
+      The dropout probability for each convolutional layer.
+    num_conv: int, default 3
+      The number of convolutional layers.
+    predicator_hidden_feats: int, default 32
+      The size for hidden representations in the output MLP predictor, default to 32.
+    n_tasks: int, default 1
+      The number of the output size, default to 1.
+    """
+    try:
+      from torch_geometric.nn import GATConv, global_mean_pool
+    except:
+      raise ValueError("This class requires PyTorch Geometric to be installed.")
+    super(GAT, self).__init__()
+    self.embedding = nn.Linear(in_node_dim, hidden_node_dim)
+    self.conv_layers = nn.ModuleList([
+        GATConv(
+            in_channels=hidden_node_dim,
+            out_channels=hidden_node_dim,
+            heads=heads,
+            concat=False,
+            dropout=dropout_rate) for _ in range(num_conv)
+    ])
+    self.pooling = global_mean_pool
+    self.fc = nn.Linear(hidden_node_dim, predicator_hidden_feats)
+    self.out = nn.Linear(predicator_hidden_feats, n_tasks)
+
+  def forward(self, data):
+    """Predict labels
+
+    Parameters
+    ----------
+    data: torch_geometric.data.Batch
+      A mini-batch graph data for PyTorch Geometric models.
+
+    Returns
+    -------
+    out: torch.Tensor
+      The output value, the shape is `(batch_size, n_out)`.
+    """
+    node_feat, edge_index = data.x, data.edge_index
+    node_feat = self.embedding(node_feat)
+
+    # convolutional layer
+    for conv in self.conv_layers:
+      node_feat = conv(node_feat, edge_index)
+
+    # pooling
+    graph_feat = self.pooling(node_feat, data.batch)
+    graph_feat = self.fc(graph_feat)
+    out = self.out(graph_feat)
+    return out
+
+
+class GATModel(TorchModel):
+  """Graph Attention Networks (GAT).
+
+  Here is a simple example of code that uses the GATModel with
+  molecules dataset.
+
+  >> import deepchem as dc
+  >> featurizer = dc.feat.MolGraphConvFeaturizer()
+  >> dataset_config = {"reload": False, "featurizer": featurizer, "transformers": []}
+  >> tasks, datasets, transformers = dc.molnet.load_tox21(**dataset_config)
```

Comment:
Wouldn't it be simpler to just write this as

```
tasks, datasets, transformers = dc.molnet.load_tox21(reload=False, featurizer=featurizer, transformers=[])
```

Reply:
I fixed!